src/libraries/Microsoft.PowerFx.Core/Binding/Binder.cs

```diff
@@ -3882,6 +3888,48 @@ public override void PostVisit(BinaryOpNode node)
                 _txb.SetIsUnliftable(node, _txb.IsUnliftable(node.Left) || _txb.IsUnliftable(node.Right));
             }
 
+            private static bool IsNoOPFirstNameComparison(BinaryOpNode node)
+            {
+                if ((node.Op == BinaryOp.Equal || 
+                    node.Op == BinaryOp.NotEqual || 
+                    node.Op == BinaryOp.Less || 
+                    node.Op == BinaryOp.LessEqual || 
+                    node.Op == BinaryOp.Greater || 
+                    node.Op == BinaryOp.GreaterEqual) && 
+                    IsNoOPFirstNameComparison(node.Left, node.Right))
+                {
+                    return true;
+                }
+
+                return false;
+            }
+
+            private static bool IsNoOPFirstNameComparison(TexlNode left, TexlNode right)
```

Comment:
>IsNoOPFirstNameComparison [](http://example.com/codeflow?start=32&length=25)

this is key change...

Reply:
NOTE: I will check the impact of the change with the PowerApps using a private build.